kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -259,6 +259,46 @@ sealed abstract class Stream[V, -S]:
         ))
     end changes
 
+    /** Transforms the stream by regrouping elements into chunks of the specified size.
+      *
+      * This operation maintains the order of elements while potentially redistributing them into new chunks. Smaller chunks may occur in
+      * two cases:
+      *   - When there aren't enough remaining elements to form a complete chunk
+      *   - When the input stream emits an empty chunk
+      *
+      * @param chunkSize
+      *   The target size for each chunk. Must be positive - negative values will be treated as 1.
+      * @return
+      *   A new stream with elements regrouped into chunks of the specified size
+      */
+    def rechunk(chunkSize: Int)(using tag: Tag[Emit[Chunk[V]]], frame: Frame): Stream[V, S] =
+        Stream.sized[V, S](chunkSize)(cs =>
+            ArrowEffect.handleState(tag, Chunk.empty[V], emit.andThen(Emit(Chunk.empty[V])))(
```

Comment:
I see now why you need to emit the empty `Chunk` at the end. I wonder if it'd be useful to change `Emit` to:

```scala
sealed trait Op[V]
sealed trait Emit[V] extends ArrowEffect[Const[Op[V]], Const[Emit.Ack]]
```

and then provide different `Op` subclasses for the different needs. Just an idea in case it's helpful.

Reply:
I don't see how that would be helpful here, but I might be missing your vision 😄

Reply:
Not sure if it's useful but I thought it was worth mentioning :) It could allow more advanced patterns by providing more information on each emission